src/Modules/OrchardCore.Commerce/Controllers/UserController.cs

```diff
@@ -86,12 +86,53 @@ public async Task<IActionResult> AddressesPost()
         return RedirectToAction(nameof(Addresses));
     }
 
-    private async Task<ContentItem> GetUserAddressesAsync(User user)
+    [HttpGet]
+    public async Task<IActionResult> Details()
+    {
+        if (User.Identity?.IsAuthenticated != true) return LocalRedirect("~/Login?ReturnUrl=~/user/details");
+        if (await _userManager.GetUserAsync(User) is not User user) return NotFound();
+
+        var userDetails = await GetUserContentItemAsync(user, UserDetails);
+        var editor = await _contentItemDisplayManager.BuildEditorAsync(
+            userDetails,
+            _updateModelAccessor.ModelUpdater,
+            isNew: false);
+
+        return View(editor);
+    }
+
+    [HttpPost]
+    [ValidateAntiForgeryToken]
+    [Route("user/details")]
+    public async Task<IActionResult> DetailsPost()
+    {
+        if (User.Identity?.IsAuthenticated != true) return LocalRedirect("~/Login?ReturnUrl=~/user/details");
+        if (await _userManager.GetUserAsync(User) is not User user) return NotFound();
+
+        var userDetails = await GetUserContentItemAsync(user, UserDetails);
+        await _contentItemDisplayManager.UpdateEditorAsync(userDetails, _updateModelAccessor.ModelUpdater, isNew: false);
+
+        if (_updateModelAccessor.ModelUpdater.ModelState.IsValid)
+        {
+            user.Put(UserDetails, userDetails);
+            _session.Save(user);
+            await _notifier.SuccessAsync(H["Your details have been updated."]);
+        }
+        else
+        {
+            var errors = _updateModelAccessor.ModelUpdater.GetModelErrorMessages();
+            foreach (var error in errors.WhereNot(string.IsNullOrEmpty)) await _notifier.ErrorAsync(H[error]);
```

Comment:
I'm pretty sure the validation errors are already localized so this should rather be `H["{0}", error]`.